crates/bevy_ptr/src/lib.rs

```diff
@@ -224,6 +238,20 @@ impl<'a> OwningPtr<'a> {
     pub fn as_ptr(&self) -> *mut u8 {
         self.0.as_ptr()
     }
+
+    /// Gets an immutable reference from this mutable reference
+    #[inline]
+    pub fn as_ref(&self) -> Ptr<'_> {
+        // SAFE: The `Owning` type's guarantees about the validity of this pointer are a superset of `Ptr` s guarantees
+        unsafe { Ptr::new(self.0) }
+    }
+
+    /// Gets an mutable reference from this
+    #[inline]
+    pub fn as_mut(&mut self) -> PtrMut<'_> {
```

Comment:
Might make more sense to call this `downgrade`, or something else to match `PtrMut::promote`.

Reply:
`demote` is the natural counterpart.

Reply:
On second thought, calling this `demote` or `downgrade` would only make sense if the method took `self` by value.